src/helm/benchmark/scenarios/oasst1_scenario.py

```diff
@@ -0,0 +1,106 @@
+                'insult': 0.00039489680784754455,
+                'threat': 4.075629112776369e-05,
+                'sexual_explicit': 2.712695459194947e-05
+            },
+            'message_tree_id': '6ab24d72-0181-4594-a9cd-deaf170242fb',
+            'tree_state': 'ready_for_export',
+            'emojis': {
+                'name': ['+1', '_skip_reply', '_skip_ranking'],
+                'count': [10, 1, 4]
+            },
+            'labels': {
+                'name': ['spam', 'lang_mismatch', 'pii', 'not_appropriate', 'hate_speech', 'sexual_content',
+                        'quality', 'toxicity', 'humor', 'creativity','violence'],
+                'value': [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.9166666666666666, 0.16666666666666666, 0.3333333333333333,
+                        0.6666666666666666, 0.0],
+                'count': [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
+            }
+        }
+        """
+
+        def get_split_instances(split: Dataset, split_tag: str):
+            """Get instances from a split (e.g. train) of the dataset"""
+            # First pass: get all initial prompts (roots of the conversation trees) and messages with role "assistant"
+            initial_prompts: Dict[str, str] = {}
+            assistant_messages: List[dict] = []
+            for msg in split:
+                assert msg["model_name"] is None
+                if msg["parent_id"] is None:
+                    assert msg["role"] == "prompter"
+                    assert msg["message_id"] not in initial_prompts
+                    initial_prompts[msg["message_id"]] = msg["text"]
+                if msg["role"] == "assistant":
+                    assistant_messages.append(msg)
+
+            # Second pass: for each initial prompt, find the corresponding assistant messages
+            prompt_responses: Dict[str, List[str]] = {}
+            for msg in assistant_messages:
+                if msg["parent_id"] in initial_prompts:
+                    prompt_responses.setdefault(msg["parent_id"], []).append(msg["text"])
+
+            # Final pass: for each initial prompt, create an instance
+            instances: List[Instance] = []
+            for msg_id, prompt in initial_prompts.items():
+                reference_texts: List[str] = []
+                if msg_id in prompt_responses:
+                    reference_texts = prompt_responses[msg_id]
+                instance = Instance(
+                    input=Input(text=prompt),
+                    references=[Reference(Output(text=ref), tags=[CORRECT_TAG]) for ref in reference_texts],
+                    split=split_tag,
+                )
+                instances.append(instance)
+            return instances
+
+        # Download the raw data from Huggingface
+        dataset: Any = load_dataset("OpenAssistant/oasst1")
+
+        # Get the instances for each split
+        return get_split_instances(dataset["train"], TRAIN_SPLIT) + get_split_instances(
```

Comment:
can we create `train_instances` and `valid_instances` as two separate variables to make this code a bit more symmetric looking / easier to read?